File: Projects/Framework-Samples/PLINQO/VB/Tracker/Source/Tracker.Tests/QueryTests/ByIntTests.vb
Imports System.Collections.Generic
Imports System.Data.Linq
Imports System.Linq
Imports NUnit.Framework
Imports Tracker.Core.Data

Namespace Tracker.Tests.QueryTests
    <TestFixture()> _
    Public Class ByIntTests
        Inherits TaskTests
        <Test()> _
        Public Sub ByTest()
            Try
                Using db = New TrackerDataContext()
                    Dim a = db.Task.ByCreatedId(userIds(1)).ToList()
                    Dim b = db.Task.ByCreatedId(userIds(2)).ToList()

                    Dim c = db.Task.ByCreatedId(userIds(1), Nothing).ToList()
                    Assert.AreEqual(a.Count, c.Count)

                    Dim d = db.Task.ByCreatedId(userIds(1), userIds(2)).ToList()
                    Assert.AreEqual(a.Count + b.Count, d.Count)
                End Using
            Catch generatedExceptionName As AssertionException
                Throw
            Catch
                Assert.Fail()
            End Try
        End Sub

        <Test()> _
        Public Sub ByNullableTest()
            Try
                Using db = New TrackerDataContext()
                    Dim a = db.Task.ByAssignedId(Nothing).ToList()
                    Dim b = db.Task.ByAssignedId(userIds(1)).ToList()
                    Dim c = db.Task.ByAssignedId(userIds(2)).ToList()

                    Dim d = db.Task.ByAssignedId(userIds(1), Nothing).ToList()
                    Assert.AreEqual(a.Count + b.Count, d.Count)

                    Dim e = db.Task.ByAssignedId(userIds(1), Nothing, userIds(2)).ToList()
                    Assert.AreEqual(a.Count + b.Count + c.Count, e.Count)
                End Using
            Catch generatedExceptionName As AssertionException
                Throw
            Catch
                Assert.Fail()
            End Try
        End Sub
    End Class
End Namespace

File: Projects/Framework-Samples/PLINQO/CSharp/Tracker/Source/Tracker.Tests/QueryTests/ByIntTests.cs
using System;
using System.Collections.Generic;
using System.Data.Linq;
using System.Linq;
using CodeSmith.Data.Linq;
using NUnit.Framework;
using Tracker.Core.Data;

namespace Tracker.Tests.QueryTests
{
    [TestFixture]
    public class ByIntTests : TestBase
    {
        [Test]
        public void ByTest()
        {

            using (var db = new TrackerDataContext { Log = Console.Out })
            {
                var a = db.Task.ByCreatedId(UserIds[1]).ToList();
                var b = db.Task.ByCreatedId(UserIds[2]).ToList();

                var c = db.Task.ByCreatedId(UserIds[1], null).ToList();
                Assert.AreEqual(a.Count, c.Count);

                var d = db.Task.ByCreatedId(UserIds[1], UserIds[2]).ToList();
                Assert.AreEqual(a.Count + b.Count, d.Count);
            }

        }

        [Test]
        public void ByNullableTest()
        {

            using (var db = new TrackerDataContext { Log = Console.Out })
            {
                var a = db.Task.ByAssignedId((int?)null).ToList();
                var b = db.Task.ByAssignedId(UserIds[1]).ToList();
                var c = db.Task.ByAssignedId(UserIds[2]).ToList();

                var d = db.Task.ByAssignedId(UserIds[1], null).ToList();
                Assert.AreEqual(a.Count + b.Count, d.Count);

                var e = db.Task.ByAssignedId(UserIds[1], null, UserIds[2]).ToList();
                Assert.AreEqual(a.Count + b.Count + c.Count, e.Count);

                var f = db.Task.ByAssignedId(ComparisonOperator.NotEquals, null).ToList();
                var g = db.Task.ByAssignedId(ComparisonOperator.Equals, null).ToList();
                
            }

        }
    }
}
